src/Hosting/Systemd/test/UserService.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Diagnostics;
+
+namespace Microsoft.Extensions.Hosting
+{
+    class UserService : IDisposable
+    {
+        private static readonly Random Random = new Random();
+        private readonly string _name;
+        private readonly string _type;
+        private readonly string _dotnetPath;
+        private readonly string _applicationAssemblyPath;
+        private readonly string _syslogIdentifier;
+
+        public UserService(string name, string type, string dotnetPath, string applicationAssemblyPath)
+        {
+            _name = name;
+            _type = type;
+            _dotnetPath = dotnetPath;
+            _applicationAssemblyPath = applicationAssemblyPath;
+            lock (Random)
+            {
+                // Unique identifier for retrieving log
+                _syslogIdentifier = "dotnet" + Random.Next();
+            }
+        }
+
+        public void Start()
+        {
+            // In case the service wasn't properly stopped in a previous test run.
+            TryStop();
+
+            WriteUnitFile();
+            RunProcess("systemctl", "daemon-reload");
```

Comment:
Isn't it supposed to be "sysctl"

Reply:
sysctl is for settings, systemctl is for interacting with systemd.